### ISSUE TITLE
setting io.cilium.proxy-visibility annotation causes LoadBalancer service to become inaccessible from other hosts

### ISSUE THREAD

**Issue by maci0** [cilium/cilium#11790]
```
maci@speedy2:~$ curl 192.168.0.211 > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1454  100  1454    0     0  1419k      0 --:--:-- --:--:-- --:--:-- 1419k
maci@speedy2:~$ echo $?
0
```

after setting the annotation on the pods

```
[root@centos ~]# kubectl  patch deployment kuard -p '{"spec":{"template":{"metadata":{"annotations":{"io.cilium.proxy-visibility": "<Ingress/8080/TCP/HTTP>,<Egress/53/UDP/DNS>"}}}}}'
deployment.apps/kuard patched
```

it just hangs and times out

```
maci@speedy2:~$ curl 192.168.0.211 > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:32 --:--:--     0
curl: (28) Failed to connect to 192.168.0.211 port 80: Connection timed out
maci@speedy2:~$ echo $?
28
```

From the node host it still works:
```
[root@centos ~]#  curl 192.168.0.211 > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1600  100  1600    0     0   781k      0 --:--:-- --:--:-- --:--:--  781k
[root@centos ~]# echo $?
0
```

cilium was installed like so:
```
helm upgrade -i cilium cilium/cilium --version 1.8.0-rc1 \
	--namespace kube-system \
	--set global.hubble.enabled=true \
	--set global.hubble.relay.enabled=true \
	--set global.hubble.ui.enabled=true \
	--set global.hubble.cli.enabled=true \
	--set global.hubble.listenAddress=":4244" \
	--set global.ipam.operator.clusterPoolIPv4PodCIDR="10.217.0.0/16" \
	--set global.hubble.ui.image.tag=latest
```

The issue appears with version `1.8.0-rc2` as well.

**Comment by brb** [comment]
@maci0 Can you paste `cilium status` output?

**Comment by joestringer** [comment]
A [sysdump](https://docs.cilium.io/en/v1.7/troubleshooting/#automatic-log-state-collection) would also help to debug this issue.

**Comment by maci0** [comment]
```
[root@centos ~]# kubectl exec -ti cilium-fnmqw -n kube-system -- cilium status
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.18 (v1.18.3) [linux/amd64]
Kubernetes APIs:        ["CustomResourceDefinition", "cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1beta1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:   Probe   (eth0)   [NodePort (SNAT, 30000-32767, XDP: NONE), HostPort, ExternalIPs, HostReachableServices (TCP)]
Cilium:                 Ok      OK
NodeMonitor:            Listening for events on 4 CPUs with 64x4096 of shared memory
Cilium health daemon:   Ok
IPAM:                   IPv4: 10/255 allocated from 10.217.0.0/24,
Controller Status:      48/48 healthy
Proxy Status:           OK, ip 10.217.0.65, 6 redirects active on ports 10000-20000
Hubble:                 Ok              Current/Max Flows: 4096/4096 (100.00%), Flows/s: 26.09   Metrics: Disabled
Cluster health:         1/1 reachable   (2020-06-03T01:39:09Z)
```

[cilium-sysdump-20200602-214303.zip](https://github.com/cilium/cilium/files/4720555/cilium-sysdump-20200602-214303.zip)

**Comment by maci0** [comment]
@joestringer @brb let me know if you need any other information

**Comment by maci0** [comment]
Experiencing the same issue on another cluster with NodePort services.

**Comment by brb** [comment]
The visibility via the L7 proxy should be using the same mechanism as regular L7 policies which is known to be not working with BPF NodePort (it is used to implement LoadBalancer type services too): https://github.com/cilium/cilium/issues/8971. For now, you could disable BPF NodePort by setting `--kube-proxy-replacement=partial` and `--enable-node-port=false`. I'm closing this issue in favor of #8971.

**Comment by brb** [comment]
@maci0 The issue might have been resolved in v1.8.1. Could you give a try?
